tests/incremental_analysis/replay_scope_patch.py

```
@@ -0,0 +1,108 @@
+        "--expect-files",
+        action="append",
+        default=[],
+        help="File path expected to belong to one of --expect-new components. May be repeated.",
+    )
+    args = parser.parse_args()
+
+    dump_path = _resolve_dump(args.dump)
+    print(f"# replaying {dump_path}")
+
+    with dump_path.open("rb") as f:
+        captured = pickle.load(f)
```

Ruff (0.15.12)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)